A32NX/html_ui/Pages/A32NX_Core/A32NX_FWC.js

```diff
@@ -286,4 +309,80 @@ class A32NX_FWC {
+        SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION_SHORT", "Bool", false);
+        SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION", "Bool", false);
+
+        // Exit when:
+        // - Landing gear down
+        // - Glide slope captured
+        const landingGearIsDown = !SimVar.GetSimVarValue("IS GEAR RETRACTABLE", "Boolean") || SimVar.GetSimVarValue("GEAR HANDLE POSITION", "Boolean");
+        const glideSlopeCaptured = SimVar.GetSimVarValue("L:GLIDE_SLOPE_CAPTURED", "bool") === 1;
+        if (landingGearIsDown || glideSlopeCaptured) {
+            return;
+        }
+
+        const currentAltitudeConstraint = SimVar.GetSimVarValue("L:A32NX_AP_CSTN_ALT", "feet");
+        // Use the constraint altitude if provided otherwise use selected altitude lock value
+        const targetAltitude = currentAltitudeConstraint && !this.getAutopilotMode() ? currentAltitudeConstraint : Simplane.getAutoPilotSelectedAltitudeLockValue();
+
+        // Exit when selected altitude is being changed
+        if (this.previousTargetAltitude !== targetAltitude) {
+            this.previousTargetAltitude = targetAltitude;
+            this._wasBellowThreshold = false;
+            this._wasAboveThreshold = false;
+            this._wasInRange = false;
+            return;
+        }
+
+        if (this.warningPressed === true) {
+            this._wasBellowThreshold = false;
+            this._wasInRange = false;
+            SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION", "Bool", false);
+            return;
+        }
+
+        const delta = Math.abs(indicatedAltitude - targetAltitude);
+
+        if (delta < 250) {
+            this._wasBellowThreshold = true;
+            this._wasAboveThreshold = false;
+        }
+
+        if (750 < delta) {
+            this._wasAboveThreshold = true;
+            this._wasBellowThreshold = false;
+        }
+
+        if (250 <= delta && delta <= 750) {
+            this._wasInRange = true;
+        }
+
+        if (250 <= delta) {
+            if (this._wasBellowThreshold) {
+                SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION", "Bool", true);
+            } else if (this._wasAboveThreshold && delta <= 750) {
+                if (SimVar.GetSimVarValue("L:XMLVAR_Autopilot_1_Status", "Bool") === 0 && SimVar.GetSimVarValue("L:XMLVAR_Autopilot_2_Status", "Bool") === 0) {
+                    SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION_SHORT", "Bool", true);
+                }
+            } else if (750 < delta && this._wasInRange) {
+                SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION", "Bool", true);
+            } else if (this._wasInRange) {
+                if (SimVar.GetSimVarValue("L:XMLVAR_Autopilot_1_Status", "Bool") === 0 && SimVar.GetSimVarValue("L:XMLVAR_Autopilot_2_Status", "Bool") === 0) {
+                    SimVar.SetSimVarValue("L:A32NX_ALT_DEVIATION_SHORT", "Bool", true);
+                }
+            }
+        }
+    }
+
+    getAutopilotMode() {
```

Comment:
To better map the semantics, I'd suggest to name it something like `hasAltitudeConstraint`.